index.js

```diff
@@ -139,26 +200,39 @@ const line = []
 
 // ***** Scope & Closures - Question 1 *****
 
+function takeATicketNumber(line){
+  line.push(1)
+  let line_counter = line.length
+  
+  // if (line.push(line_length) === 0){
+  //   line.push(1)
+  // }
+  
+
+  return `Welcome. You are number ${line.length} in line.`
+  line.unshift(line_counter)
+}
```

Comment:
Hmm, not exactly. Take a look at the solution and see if you can trace what it's doing